adam-core/src/main/java/org/bdgenomics/adam/io/IndexedBamInputFormat.scala

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package htsjdk.samtools
+
+import htsjdk.samtools.util.BlockCompressedFilePointerUtil
+import org.apache.hadoop.fs.{ FileSystem, Path }
+import org.apache.hadoop.mapreduce.JobContext
+import org.apache.hadoop.mapreduce.RecordReader
+import org.apache.hadoop.mapreduce.InputSplit
+import org.apache.hadoop.mapreduce.TaskAttemptContext
+import org.apache.hadoop.mapreduce.lib.input.FileSplit
+import org.seqdoop.hadoop_bam.BAMInputFormat
+import org.seqdoop.hadoop_bam.FileVirtualSplit
+import org.seqdoop.hadoop_bam.SAMRecordWritable
+import org.bdgenomics.adam.models.ReferenceRegion
+import org.bdgenomics.adam.rdd.ADAMContext._
+import org.bdgenomics.adam.io.BAMFilteredRecordReader
+import org.apache.hadoop.io.LongWritable
+import java.io.File
+import scala.collection.mutable
+
+object IndexedBamInputFormat {
+
+  var optFilePath: Option[Path] = None
+  var optIndexFilePath: Option[Path] = None
+  var optViewRegion: Option[ReferenceRegion] = None
+  var optDict: Option[SAMSequenceDictionary] = None
+
+  def apply(filePath: Path, indexFilePath: Path, viewRegion: ReferenceRegion, dict: SAMSequenceDictionary) {
+    optFilePath = Some(filePath)
+    optIndexFilePath = Some(indexFilePath)
+    optViewRegion = Some(viewRegion)
+    optDict = Some(dict)
+  }
+
+}
+
+class IndexedBamInputFormat extends BAMInputFormat {
+
+  override def createRecordReader(split: InputSplit, ctx: TaskAttemptContext): RecordReader[LongWritable, SAMRecordWritable] = {
+    val rr: RecordReader[LongWritable, SAMRecordWritable] = new BAMFilteredRecordReader()
+    assert(IndexedBamInputFormat.optViewRegion.isDefined)
+    BAMFilteredRecordReader(IndexedBamInputFormat.optViewRegion.get)
```

Comment:
What does this line do? Since it is an apply, I'm a little surprised that the return value isn't being used.